### ISSUE TITLE
how to run it or get it working?

### ISSUE THREAD

**Issue by lfatty** [Etraud123/PyCheckmarx#1]

**Comment by dvolvox** [comment]
Hi sorry for the lack of documentation, I'm still working on it.

1# - Git clone the project
2# - Create a file in the etc folder called: config.json 
3# - use the config.json.template to place the data in the config.json with the credentials and the endpoint of the API. The endpoint should be: http://x.x.x.x/Cxwebinterface/CxWsResolver.asmx?wsdl".
4# - After installing all dependencies: python APICheckmarx.py
5# - An endpoint is created in localhost port 5000 (change in APICheckmarx.py)
6# - Queries:

> 0.0.0.0:5000/APICX/getProjectScannedDisplayData/
> 0.0.0.0:5000/APICX/getScanInfoForAllProjects/
> 0.0.0.0:5000/APICX/getPresetList/
> 0.0.0.0:5000/APICX/getConfigurationList/
> 0.0.0.0:5000/APICX/getAssociatedGroups/

If you want to incorporate the script in other just place the PyCheckmarx.py file in your project and

import PyCheckmarx

You can code using the API:
e.g:
tmp = PyCheckmarx()
print tmp.filterProjectScannedDisplayData(20004)
print tmp.getSupressedIssues("1000004")

If you follow the flow of the code its pretty easy to understand how it works. Let me know if you need any help.

**Comment by lfatty** [comment]
@Etraud123 
Thanks for clarifying the steps. Also it be will nice if you could add few functionalities such as creating new project and scanning that project and seeing the result of that scan which may help developers big time.

**Comment by dvolvox** [comment]
No problem. 

You are right. But this was just a small project and the purpose of it was to integrate Checkmarx in a Continuous Delivery environment (get results and scan dates and projects) and since they didn't have any automation I did my own. They have upgraded a bit check:
https://checkmarx.atlassian.net/wiki/display/KC/OData
But either way I wanted more info.

**Comment by lfatty** [comment]
Yes indeed. Are you guys using the checkmarx jenkins plugin?https://checkmarx.atlassian.net/wiki/display/KC/Configuring+a+Project

**Comment by lfatty** [comment]
Here what i found on checkmarx wiki.

internal void Main(string [] args)
{
   String sessionID = args[0];
   CxSDKWebServiceSoapClient cxSDKProxy = new CxSDKWebServiceSoapClient();
 
   ProjectSettings projSettings = new ProjectSettings();
 
   //The project full name
   projSettings.ProjectName = @"CxServer\SP\Company\NewScanProject";
 
   //Set the query preset to 'default' preset (ID = 7)
   projSettings.PresetID = 7;
 
   //Set the source files encoding, English = 1
   projSettings.ScanConfigurationID = 1;
 
   SourceCodeSettings sourceCodeSettings = new SourceCodeSettings();
 
   //Set the source code location to be local
   sourceCodeSettings.SourceOrigin = SourceLocationType.Local;
 
   //Set the zipped file and put its contents into byte array
   sourceCodeSettings.PackagedCode = new LocalCodeContainer();
   sourceCodeSettings.PackagedCode.FileName = @"C:\Server\Sources.zip";
   sourceCodeSettings.PackagedCode.ZippedFile = File.ReadAllBytes(sourceCodeSettings.PackagedCode.FileName);
 
  
   CliScanArgs scanArgs = new CliScanArgs();
   scanArgs.PrjSettings = projSettings;
   scanArgs.SrcCodeSettings = sourceCodeSettings;
 
   //The scan is public for all users
   scanArgs.IsPrivateScan = false;
 
   //Scan all sources, not just changed sources
   scanArgs.IsIncremental = false;
 
   CxWSResponseRunID response = cxSDKProxy.Scan(sessionID, scanArgs);
   ScanSucceded = response.IsSuccesfull;
   RunID = response.RunId
   PrjctID = response.ProjectID
}

public CxWSResponseScanStatus GetStatusOfSingleScan(
   string sessionID,
   string runId
);

internal void Main(string [] args)
{
    String sessionID = args[0];
    CxSDKWebServiceSoapClient cxSDKProxy = new CxSDKWebServiceSoapClient();
 
    //the scan's known identifier
    string scanRunID = "d95d56a2-2d8c-4e61-8146-0822213549f8";
 
    CxWSResponseScanStatus response = cxSDKProxy.GetStatusOfSingleScan(sessionID, scanRunID);
 
    //store the scan's current state
    CurrentStatus = response.CurrentStatus;
}

public CxWSCreateReportResponse CreateScanReport(
   string sessionID,
   CxWSReportRequest reportRequest
);

internal void Main(string [] args)
{
    String sessionID = args[0];
    CxSDKWebServiceSoapClient cxSDKProxy = new CxSDKWebServiceSoapClient();
 
    CxWSReportRequest request = new CxWSReportRequest ();
 
    //set the report type to be PDF
    request.Type = CxWSReportType.PDF;
 
    //report should be for scan ID 256
    request.ScanID = 256;
 
    CxWSCreateReportResponse response = cxSDKProxy.CreateScanReport(sessionID, request);
    ReportID =response.ID;
}

**Comment by dvolvox** [comment]
Jenkins Plugin would be a nice idea, but a checkmarx scan takes more than 15 hours for our projects and we can't take that long to build. That's why we couldn't use that. 

Yes I have all their samples.

**Comment by lfatty** [comment]
I used the plugin and include jenkins on the build flow. How do you get developers use checkmarx on the regular basis?

**Comment by schattenbaum** [comment]
I just recently came across your project and liked your idea.
However, taking it out of your discussion, I wanted to give you a short hint. ;)
The Cx Jenkins plugin can be configured to scan in an asynchronous mode. This means, that the plugin will simply take your sources, zip them, send them to Cx for a scan, and the build will continue and therefore does not have to wait the ~15h. Therefore, maybe this plugin could still be of use to you. Nevertheless, keep up the good work, as I might want to use it for a HipChat integration of Cx. ;)

**Comment by dvolvox** [comment]
@lfatty  The devs use the web interface to review the issues.
@schattenbaum hmm I did not know that, but whats the difference from scheduling on checkmarx for a weekly bases scan? We have a massive amount of build per week

**Comment by schattenbaum** [comment]
There is not really a huge difference: On the one side, you may execute a scan whenever a build is triggered (attention: this could lead to a massive amount of scans, depending on your build strategy. Also be aware of the fact, that if your Cx scan, for whatever reason, crashes, your build job will fail!) and manage the build and the scan creation on one system (Jenkins),
on the other side are you completely free when you want to scan what projects and even if Jenkins should be out of order (I hope this won’t ever be the case ;)) your scans could still be triggered.

You are probably already using the one thing that works best for you. I still wanted to inform you about the possibility, as I sometimes are happy as well, if I get a hit on a given topic. ;)

What might actually could be of you interest, is that the Cx Jenkins plugin allows you to set thresholds (amount of highs, mediums and lows). Once this threshold is exceeded, the build is marked as unstable. This is probably not what you want for all of your builds, but you might want to use it for some highly important resp. critical build jobs that should not have any high or even medium Cx findings in them.
